apps/partners-reference/pages/applications.tsx

```diff
@@ -1,22 +1,126 @@
-import { Component } from "react"
+import { Component, useContext } from "react"
+import useSWR from "swr"
+
 import Head from "next/head"
-import { PageHeader, MetaTags, t } from "@bloom-housing/ui-components"
+import { PageHeader, MetaTags, ApiClientContext, t } from "@bloom-housing/ui-components"
+import { Application } from "@bloom-housing/core"
 import Layout from "../layouts/application"
 
-export default class extends Component {
-  public render() {
-    const metaDescription = t("pageDescription.welcome", { regionName: t("region.name") })
-    const metaImage = "" // TODO: replace with hero image
-
-    return (
-      <Layout>
-        <Head>
-          <title>{t("nav.siteTitle")}</title>
-        </Head>
-        <MetaTags title={t("nav.siteTitle")} image={metaImage} description={metaDescription} />
-        <PageHeader>List of Applications will go here.</PageHeader>
-        <div className="homepage-extra"></div>
-      </Layout>
+import { AgGridReact } from "ag-grid-react"
+import { Listing } from "@bloom-housing/backend-core"
+
+function ListingsData() {
+  const fetcher = (url) => fetch(url).then((r) => r.json())
+  const { data, error } = useSWR("http://localhost:3100", fetcher)
```

Comment:
Yeah, my vision is that we should consolidate all of the client->backend logic with configurable baseUrl, automagic authentication, centralized retry/failure logic, etc. I just figured I'd try the traditional SWR implementation here first since it's been a public call to date. Per your comment above, we should probably make refactoring to that future a separate issue/PR.

Reply:
Actually, now re-reading your comment, I agree we can do that first part of the refactor here, and then think about moving the codegen stuff later...